src/VirtoCommerce.ProfileExperienceApiModule.Data/Commands/RegisterCompanyCommandHandler.cs

```diff
@@ -0,0 +1,194 @@
+                await _memberService.SaveChangesAsync(new Member[] { owner });
+                aggregate.Owner = owner;
+
+                var maintainerRole = await _roleManager.FindByIdAsync(MaintainerRoleId);
+                if (maintainerRole == null)
+                {
+                    throw new ArgumentException($"Organization maintainer role with id {MaintainerRoleId} not found");
+                }
+
+                account.StoreId = request.StoreId;
+                account.Status = contactStatus;
+                account.UserType = UserType;
+                account.MemberId = owner.Id;
+                account.Roles = new List<Role> { maintainerRole };
+                aggregate.AccountCreationResult = await CreateAccountAsync(account);
+
+                if (!aggregate.AccountCreationResult.Succeeded)
+                {
+                    await RollBackMembersCreationAsync(aggregate);
+                    return aggregate;
+                }
+
+                aggregate.Account = account;
+
+                if (company != null)
+                {
+                    await SendNotificationAsync(account.Email, store.Email, company.Name);
+                }
+
+                return aggregate;
+            }
+            catch (Exception)
+            {
+                await RollBackMembersCreationAsync(aggregate);
+                throw;
+            }
+        }
+
+        private async Task RollBackMembersCreationAsync(RegisterCompanyAggregate aggregate)
+        {
+            var ids = new[] { aggregate.Company?.Id, aggregate.Owner?.Id }
+                .Where(x => x != null)
+                .ToArray();
+            await _memberService.DeleteAsync(ids);
+
+            aggregate.Company = null;
+            aggregate.Owner = null;
+        }
+
+        private async Task<IdentityResult> CreateAccountAsync(ApplicationUser account)
+        {
+            var result = default(IdentityResult);
+
+            using var userManager = _userManagerFactory();
+            if (account.Password.IsNullOrEmpty())
+            {
+                result = await userManager.CreateAsync(account);
+            }
+            else
+            {
+                result = await userManager.CreateAsync(account, account.Password);
+            }
+
+            if (result.Succeeded)
+            {
+                var user = await userManager.FindByNameAsync(account.UserName);
+
+                await _storeNotificationSender.SendUserEmailVerificationAsync(user);
+            }
+
+            return result;
+        }
+
+        private async Task SetDynamicPropertiesAsync(IList<DynamicPropertyValue> dynamicProperties, IHasDynamicProperties entity)
+        {
+            if (dynamicProperties?.Any() ?? false)
+            {
+                await _dynamicPropertyUpdater.UpdateDynamicPropertyValues(entity, dynamicProperties);
+            }
+        }
```

Comment:
No reason for awaiting here.
```suggestion
        private Task SetDynamicPropertiesAsync(IList<DynamicPropertyValue> dynamicProperties, IHasDynamicProperties entity)
        {
            if (dynamicProperties?.Any() ?? false)
            {
                return _dynamicPropertyUpdater.UpdateDynamicPropertyValues(entity, dynamicProperties);
            }
            
            return Task.CompletedTask;
        }
```